syntax = "proto3";

package wpi.proto;

option java_package = "edu.wpi.first.math.proto";

message ProtobufTranslation2d {
  double x_meters = 1;
  double y_meters = 2;
}

message ProtobufRotation2d {
  double radians = 1;
}

message ProtobufPose2d {
  ProtobufTranslation2d translation = 1;
  ProtobufRotation2d rotation = 2;
}

message ProtobufTransform2d {
  ProtobufTranslation2d translation = 1;
  ProtobufRotation2d rotation = 2;
}

message ProtobufTwist2d {
  double dx_meters = 1;
  double dy_meters = 2;
  double dtheta_radians = 3;
}
